test/ventilation_test.c

```diff
@@ -29,16 +32,60 @@ int startVentilation_afterOneInspiration_openInspiratoryValve() {
 
   //reinitialize the system to be sure that the valve wasn't previously open
   close_inspiratory_valve();
+  reset_to_inspiration_start();
   
   start_inspiration();
 
   int inspiratory_valve_status = get_inspiratory_valve_status();
   assertTrue(inspiratory_valve_status == VALVE_OPEN);
 }
 
+int startVentilation_afterOneInspiration_setsVentilationPhaseToInspiration() {
+  //initialize the function get_current_time
+  int* times = (int*) malloc(1*sizeof(int)); 
+  times[0] = 260;
+  set_current_time(times, 1);
+
+  //initialize the function get_respiratory_rate
+  uint32_t* RR = (uint32_t*) malloc(1*sizeof(uint32_t)); 
+  RR[0] = 7500000;
+  status_t* status= (status_t*) malloc(1*sizeof(status_t));
+  status[0] = STATUS_OK;
+  set_respiratory_rate(RR, status, 1);
+
+  reset_to_inspiration_start();
+
+  start_inspiration();
+
+  int ventilation_phase = get_ventilation_phase();
+  assertTrue(ventilation_phase == 1);
+}
+
+int startVentilation_afterTwoInspirations_updatesStartCurrentBreathCycle() {
+  //initialize the function get_current_time
+  int* times = (int*) malloc(1*sizeof(int)); 
+  int time = 390;
+  times[0] = time;
+  set_current_time(times, 1);
+
+  //initialize the function get_respiratory_rate
+  uint32_t* RR = (uint32_t*) malloc(1*sizeof(uint32_t)); 
+  RR[0] = 7500000;
+  status_t* status= (status_t*) malloc(1*sizeof(status_t));
+  status[0] = STATUS_OK;
+  set_respiratory_rate(RR, status, 1);
+
+  reset_to_inspiration_start();
+
+  start_inspiration();
+
+  int time_of_start_current_breath_cycle = get_start_current_breath_cycle();
+  assertTrue(time_of_start_current_breath_cycle == time);
+}
+
 int startVentilation_duringInspiration_doesNotopenInspiratoryValve() {
```

Comment:
I find this test name confusing as well... during inspiration, do we not want the valve to be opened? This name does not make sense.

I think what is meant is: "the valve is not re-opened after inspiration has started". Yes? If so, please adjust the name of the test.

Reply:
OK, reading the test code below, I am extra confused... please clarify.